src/editor/Editor.js

```diff
@@ -1,260 +1,170 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import Classable from '../mixins/classable';
-import BlockStyleControls from './BlockStyleControls';
-import InlineStyleControls from './InlineStyleControls';
-import immutable from 'immutable';
-import classnames from 'classnames';
 import ThemeService from '../styles/ChamelThemeService';
-import {stateToHTML} from 'draft-js-export-html';
-import {stateFromHTML} from 'draft-js-import-html';
-import {Editor, EditorState, RichUtils} from 'draft-js';
+import ContentHtml from './ContentHtml';
+import ContentSource from './ContentSource';
+
+// Types of content views
+const HTML_VIEW = 1;
+const SOURCE_VIEW = 2;
 
-// Custom overrides for "code" style.
-const styleMap = {
-  CODE: {
-    backgroundColor: 'rgba(0, 0, 0, 0.05)',
-    fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
-    fontSize: 16,
-    padding: 2,
-  },
-};
 
 /**
  * Chamel Editor component for editing rich text and source code
  */
 class ChamelEditor extends React.Component {
 
-  /**
-   * Set accepted properties
-   */
-  static propTypes = {
     /**
-     * The callback function used when user changes the content of the editor
-     *
-     * @type {function}
+     * Set accepted properties
      */
-    onChange: React.PropTypes.func,
+    static propTypes = {
+        /**
+         * The callback function used when user changes the content of the editor
+         *
+         * @type {function}
+         */
+        onChange: React.PropTypes.func,
+
+        /**
+         * The callback function used when user looses the focus of the editor
+         *
+         * @type {function}
+         */
+        onBlur: React.PropTypes.func,
+
+        /**
+         * The initial value of the content editor
+         *
+         * @type {string}
+         */
+        value: React.PropTypes.string,
+
+        /**
+         * Determine what is the intial content view to be displayed
+         *
+         * @type {string}
+         */
+        contentView: React.PropTypes.oneOf([HTML_VIEW, SOURCE_VIEW])
```

Comment:
The constants HTML_VIEW and SOURCE_VIEW are not being exported, so consumers of this will have to enter 1 or 2 when they render the editor, which is a strange input. I would rather make the constants string like 'html' and 'source' and then allow the user to use text. Either that or expose the constants in Editor/types.js, but I don't think we need to go that far.